tls/s2n_x509_validator.c

```diff
@@ -210,6 +212,29 @@ static uint8_t s2n_verify_host_information(struct s2n_x509_validator *validator,
             size_t name_len = (size_t) ASN1_STRING_length(current_name->d.ia5);
 
             verified = conn->verify_host_fn(name, name_len, conn->data_for_verify_host);
+        } else if (current_name->type == GEN_IPADD) {
+            /* try to validate an IP address if it's in the subject alt name. */
+            const unsigned char *ip_addr = current_name->d.iPAddress->data;
+            size_t ip_addr_len = (size_t)current_name->d.iPAddress->length;
+
+            char address[INET6_ADDRSTRLEN + 1] = {0};
+            const char *address_ptr = NULL;
+            if (ip_addr_len == 4) {
+                struct sockaddr_in s = {0};
+                s.sin_family = AF_INET;
+                memcpy(&s.sin_addr.s_addr, ip_addr, ip_addr_len);
+                address_ptr = inet_ntop(AF_INET, &s.sin_addr, address, (socklen_t) sizeof(address));
```

Comment:
inet_ntop can still fail with EAFNOSUPPORT, what should we do in that case? docs say we need to check for INADDR_NONE.

Reply:
inet_ntop returns NULL if there's an error and we check that it's not null before proceeding. Worst case, we just don't validate that name.